### PR TITLE
[prometheus-node-exporter] Fix very slow GKE cluster upgrades

### PR DESCRIPTION
**Pull request by gw0**
#### What this PR does / why we need it:
I noticed that upgrading the nodes in a GKE cluster that contains `kube-prometheus-stack` with `nodeExporter` enabled is incredibly slow. Especially, if you use a rolling upgrade with 1 extra node it takes around 40 min per node (should take <5 min). It turns out the upgrade process is waiting for `prometheus-node-exporter` pods to get deleted, but these have tolerations to prevent that.

#### Special notes for your reviewer:
I was monitoring an upgrade process on GKE recently. First all pods got rescheduled to other nodes, but then each node got stuck in the below state for 40 min before a timeout decided to decommission the node. If I deleted the `monitoring-prometheus-node-exporter-tabcv` pod manually, it decommissioned the node straight away.

```bash
## while waiting for the node upgrade
$ kubectl describe node/gke-node-pool-8e12a34c-0hab
Name:               gke-node-pool-8e12a34c-0hab
Roles:              <none>
Labels:             beta.kubernetes.io/arch=amd64
                    beta.kubernetes.io/instance-type=n1-standard-2
                    beta.kubernetes.io/os=linux
                    cloud.google.com/gke-nodepool=node-pool
                    cloud.google.com/gke-os-distribution=cos
                    failure-domain.beta.kubernetes.io/region=us-west1
                    failure-domain.beta.kubernetes.io/zone=us-west1-a
                    kubernetes.io/arch=amd64
                    kubernetes.io/hostname=gke-node-pool-8e12a34c-0hab
                    kubernetes.io/os=linux
Annotations:        container.googleapis.com/instance_id: 1234567890
                    gke-current-operation:
                      operation_type:  UPGRADE_NODES
                      lock_end_timestamp:  1607611342
                      operation_name:  "operation-123456-1234567890"
                    node.alpha.kubernetes.io/ttl: 0
                    node.gke.io/last-applied-node-labels: cloud.google.com/gke-nodepool=node-pool,cloud.google.com/gke-os-distribution=cos
                    volumes.kubernetes.io/controller-managed-attach-detach: true
CreationTimestamp:  Fri, 06 Nov 2020 22:57:16 +0000
Taints:             node.kubernetes.io/unschedulable:NoSchedule
Unschedulable:      true
Non-terminated Pods:         (5 in total)
  Namespace                  Name                                       CPU Requests  CPU Limits  Memory Requests  Memory Limits  AGE
  ---------                  ----                                       ------------  ----------  ---------------  -------------  ---
  kube-system                fluentd-gke-1abc5                          100m (5%)     1 (51%)     200Mi (3%)       500Mi (8%)     33d
  kube-system                gke-metrics-agent-tabc7                    3m (0%)       0 (0%)      50Mi (0%)        50Mi (0%)      25h
  kube-system                kube-proxy-gke-node-pool-8e12a34c-0hab     100m (5%)     0 (0%)      0 (0%)           0 (0%)         33d
  kube-system                prometheus-to-sd-gdabc                     0 (0%)        0 (0%)      0 (0%)           0 (0%)         25h
  monitoring                 monitoring-prometheus-node-exporter-tabcv  0 (0%)        0 (0%)      0 (0%)           0 (0%)         31d
Events:
  Type    Reason              Age   From                                                          Message
  ----    ------              ----  ----                                                          -------
  Normal  NodeNotSchedulable  55s   kubelet, gke-node-pool-8e12a34c-0hab  Node gke-node-pool-8e12a34c-0hab status is now: NodeNotSchedulable
```

Probable cause:
https://github.com/prometheus-community/helm-charts/blob/c31e539d324256a87cdafeac3f15088e3225f693/charts/prometheus-node-exporter/values.yaml#L98-L100

Probably best solution based on reading [the Kubernetes Cluster Autoscaler docs](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#what-types-of-pods-can-prevent-ca-from-removing-a-node):

> What types of pods can prevent CA from removing a node?
> ...
> Unless the pod has the following annotation (supported in CA 1.0.3 or later):
> "cluster-autoscaler.kubernetes.io/safe-to-evict": "true"

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by monotek — DISMISSED** [review]

**Comment by gw0** [comment]
It seems to me that the reviewing/approving process with 2 reviewers is not working?

**Comment by monotek** [comment]
@scottrigby 
could you please jump in and review / merge?

File: charts/prometheus-node-exporter/values.yaml [hunk 1]
```diff
@@ -84,7 +84,9 @@ affinity: {}
 #                 - target-host-name
 
 # Annotations to be added to node exporter pods
-podAnnotations: {}
+podAnnotations:
+  # Fix for very slow GKE cluster upgrades
+  cluster-autoscaler.kubernetes.io/safe-to-evict: "true"
```
Comment (scottrigby) [inline charts/prometheus-node-exporter/values.yaml:89]: I don't understand why this PR exists. Are you having trouble passing this map to the existing `annotations` helm value? It appears you can already do this.

**Comment by scottrigby** [comment]
> It seems to me that the reviewing/approving process with 2 reviewers is not working?

The reason is CI is looking for approval by one of the maintainers of this chart. In this repo PROCESSES.md doc, we state that a repo admin may review/merge a PR for a chart if that chart's maintainers do not respond within 2 weeks. This falls into that category. 

As a repo admin I have now made an initial review - please see the question I asked as part of my review.

File: charts/prometheus-node-exporter/values.yaml (hunk 1 above)
Comment (gw0) [reply charts/prometheus-node-exporter/values.yaml:89]: The purpose of this PR is to provide a new default setting that fixes an issue that all GKE are experiencing (most without knowing why). It is in `values.yaml` to allow everyone to override it if needed.

**Review by vsliouniaev — DISMISSED** [review]
Seems like you need to bump the version in the chart

**Comment by gw0** [comment]
@vsliouniaev Thx for bumping the version.

**Review by vsliouniaev — APPROVED** [review]

**Pull request merged**